packages/charting/src/components/DonutChart/Arc/Arc.tsx

```diff
@@ -58,11 +53,16 @@ export class Arc extends React.Component<IArcProps, { isCalloutVisible: boolean
     }
   }
 
-  private _hoverOff(): void {
+  private _hoverOff = (): void => {
     this.props.hoverLeaveCallback!();
-  }
+  };
 
   private _redirectToUrl(href: string | undefined): void {
     href ? (window.location.href = href) : '';
   }
 }
+
+function _updateChart(newProps: IArcProps): void {
+  newProps.arc.innerRadius(newProps.innerRadius);
```

Comment:
I know you're just moving code from one place to another... but this is a code smell.

@RajeshGoriga @samiyaakhtar props should not be mutated, ever. If you need to process them, you should process a copy of the props and set it in state (or, in the new API `getDerivedStateFromProps`, return the state values derived from the new props.)